cartridge.lua

```diff
@@ -461,9 +462,27 @@ local function cfg(opts, box_opts)
 
     membership.set_encryption_key(cluster_cookie.cookie())
     membership.set_payload('alias', opts.alias)
-    local ok, estr = membership.probe_uri(membership.myself().uri)
-    if not ok then
-        return nil, CartridgeCfgError:new('Can not ping myself: %s', estr)
+
+    local probe_uri_opts, err = argparse.get_opts({probe_uri_timeout = 'number'})
+    if err ~= nil then
+        return nil, err
+    end
+
+    local delay = require('membership.options').ACK_TIMEOUT_SECONDS
+    local deadline = fiber.time() + (probe_uri_opts.probe_uri_timeout or 0)
+
+    while true do
+        local next_wakeup = fiber.time() + delay 
+        local ok, estr = membership.probe_uri(membership.myself().uri)
+        local now = fiber.time()
+        if ok then
+            break
+        elseif now >= deadline then
+            return nil, CartridgeCfgError:new('Can not ping myself: %s', estr)
```

Comment:
"can not ping myself" because of what? consider adding something like "deadline exceeded" to log message

Reply:
because of `%s, estr`